bundle/orchestrate/dags/meltano.py

```diff
@@ -32,75 +32,152 @@
-        tags.append(schedule["loader"])
-    if schedule["transform"] == "run":
-        tags.append("transform")
-    elif schedule["transform"] == "only":
-        tags.append("transform-only")
-
-    # from https://airflow.apache.org/docs/stable/scheduler.html#backfill-and-catchup
-    #
-    # It is crucial to set `catchup` to False so that Airflow only create a single job
-    # at the tail end of date window we want to extract data.
-    #
-    # Because our extractors do not support date-window extraction, it serves no
-    # purpose to enqueue date-chunked jobs for complete extraction window.
-    dag = DAG(
-        dag_id,
-        tags=tags,
-        catchup=False,
-        default_args=args,
-        schedule_interval=schedule["interval"],
-        max_active_runs=1,
-    )
 
-    elt = BashOperator(
-        task_id="extract_load",
-        bash_command=f"cd {project_root}; {meltano_bin} schedule run {schedule['name']}",
-        dag=dag,
+        # register the dag
+        globals()[dag_id] = dag
+        logger.info(f"DAG created for schedule '{schedule['name']}'")
+
+
+def _meltano_job_generator(schedules):
+    """Generate dag's for each task within a Meltano scheduled job.
+
+    Args:
+        schedules (list): List of Meltano scheduled jobs.
+    """
+    for schedule in schedules:
+        if not schedule.get("job"):
+            logger.info(
+                f"No DAG's created for schedule '{schedule['name']}'. It was passed to job generator but has no job."
+            )
+            continue
+        if not schedule["cron_interval"]:
+            logger.info(
+                f"No DAG created for schedule '{schedule['name']}' because its interval is set to `@once`."
+            )
+            continue
+
+        base_id = f"meltano_{schedule['name']}_{schedule['job']['name']}"
+        common_tags = DEFAULT_TAGS.copy()
+        common_tags.append(f"schedule:{schedule['name']}")
+        common_tags.append(f"job:{schedule['job']['name']}")
+        interval = schedule["cron_interval"]
+
+        for idx, task in enumerate(schedule["job"]["tasks"]):
+            logger.info(
+                f"Considering task '{task}' of schedule '{schedule['name']}': {schedule}"
+            )
+            args = DEFAULT_ARGS.copy()
+            args["start_date"] = datetime.utcnow()
+            dag_id = f"{base_id}_task{idx}"
+            task_tags = common_tags.copy()
+
+            dag = DAG(
+                dag_id,
+                description=f"Meltano run task[{idx}]: '{task}'",
+                tags=task_tags,
+                catchup=False,
+                default_args=args,
+                schedule_interval=interval,
+                max_active_runs=1,
+            )
+
+            if isinstance(task, Iterable) and not isinstance(task, str):
+                run_args = " ".join(task)
+            else:
+                run_args = task
+
+            elt = BashOperator(
+                task_id=f"run_task_{idx}",
```

Comment:
I wasn't sure what task_id is really used for, and going by what we had originaly it doesn't need to be unique. So I just rolled with this 🤷‍♂️